aten/src/ATen/core/dispatch/OperatorEntry.cpp

```diff
@@ -211,11 +213,20 @@ std::pair<const AnnotatedKernel&, const char*> OperatorEntry::computeDispatchTab
   }
 }
 
+void OperatorEntry::updateDispatchTableEntry_(const c10::Dispatcher& dispatcher, DispatchKey dispatch_key) {
+  auto dispatch_ix = static_cast<uint8_t>(dispatch_key);
+  dispatchTable_[dispatch_ix] = computeDispatchTableEntry(dispatcher, dispatch_key);
+  dispatchKeyExtractor_.setOperatorHasFallthroughForKey(dispatch_key, dispatchTable_[dispatch_ix].isFallthrough());
+}
+
 void OperatorEntry::updateDispatchTable_(const c10::Dispatcher& dispatcher, DispatchKey dispatch_key) {
   for (auto k : c10::getRuntimeDispatchKeys(dispatch_key)) {
-    auto dispatch_ix = static_cast<uint8_t>(k);
-    dispatchTable_[dispatch_ix] = computeDispatchTableEntry(dispatcher, k);
-    dispatchKeyExtractor_.setOperatorHasFallthroughForKey(k, dispatchTable_[dispatch_ix].isFallthrough());
+    updateDispatchTableEntry_(dispatcher, k);
+  }
+  // Registering to backend key might affect computed entry at its Autograd backend key due to 2.2.
+  DispatchKey autograd_key = getAutogradKeyFromBackend(dispatch_key);
```

Comment:
What if `autograd_key` is undefined?

Reply:
autograd_key is default to AutogradOther if it misses all the switch conditions ;)